stubs/regex/regex/_regex.pyi

```diff
@@ -0,0 +1,184 @@
+from typing import AnyStr, Callable, Generic, Iterator, NoReturn, TypeVar, overload
+
+_T = TypeVar("_T")
+
+class Pattern(Generic[AnyStr]):
+
+    pattern: AnyStr
+    flags: int
+    groups: int
+    groupindex: dict[str, int]
+    named_lists: dict[str, frozenset[AnyStr]]
```

Comment:
Is `Mapping` preferred over `dict` in both these cases? (that's what `typing.pyi` does for the `re` equivalent)

Reply:
It prevents modifying `named_lists` accidentally. Consistency with `re` would be nice..